### PR TITLE
Adding filter definition

### PR DESCRIPTION
**Pull request by tsurdilo**
The document was missing filter definition which is described in the "Filter Mechanism" section. This adds it.

### CODE REVIEW

**Comment by tsurdilo** [comment]
ill fix merge conflicts once a decision on this has been made (before merging)

**Comment by tsurdilo** [comment]
@ruromero rebased

**Comment by ruromero** [comment]
@cathyhongzhang any comments?

**Comment by cathyhongzhang** [comment]
@ruromero Thanks for checking with me. I am in meetings and will find time to review it this afternoon.

File: workflow/spec/spec.md [hunk 1]
```diff
@@ -482,15 +496,18 @@ as well as define parameters (key/value pairs).
             "description": "Actions Definitions",
             "items": {
                 "type": "object",
-                "$ref": "#definitions/action"
+                "$ref": "#/definitions/action"
             }
         },
+        "filter": {
+          "$ref": "#/definitions/filter"
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: If I read the syntax correctly, it does not seem to allow applying different filters to different actions. The original spec allows that.

File: workflow/spec/spec.md [hunk 2]
```diff
@@ -911,6 +940,42 @@ The elements of the output array need not be of the same type.
...
+      "type": "string",
+      "description": "Specify result JSON node of Action Output as JSONPath"
+    },
+    "output-path": {
+      "type": "string",
+      "description": "Specify output data of State or Action as JSONPath"
+    }
+  },
+  "required": ["input-path", "result-path", "output-path"]
+}
```
Comment (cathyhongzhang) [inline workflow/spec/spec.md]: All three are required, but for an event, we only need input-path filter. Could you clarify? Thanks

File: workflow/spec/spec.md (hunk 1 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: @cathyhongzhang each action definition includes a filter property. so the functionality you mention is preserved. The operation state itself also needs a filter definitions which is what you are pointing at.

File: workflow/spec/spec.md (hunk 2 above)
Comment (tsurdilo) [reply workflow/spec/spec.md]: good point. fixed

**Comment by tsurdilo** [comment]
i will rebase this in case the camel case props pr gets pushed before this

File: workflow/spec/spec.md (hunk 1 above)
Comment (cathyhongzhang) [reply workflow/spec/spec.md]: I see. Thanks for the clarification. I thought this filter is for the action.

File: workflow/spec/spec.md (hunk 2 above)
Comment (cathyhongzhang) [reply workflow/spec/spec.md]: LGTM

**Comment by ruromero** [comment]
I will merge it then once #96 is merged

**Pull request merged**